app/assets/javascripts/add-reduction.js

```diff
@@ -10,59 +10,69 @@
-    if (chosenReduction.monthsToExpiry !== null) {
-      if (Number(startMonth) > 0 && Number(startMonth) < 13) {
-        var startDate = new Date(Number(document.getElementById("start-year").value),
-                                 Number(document.getElementById("start-month").value),
-                                 Number(document.getElementById("start-day").value))
-        var endDate = new Date(startDate.setMonth(startDate.getMonth() + chosenReduction.monthsToExpiry))
-        readonly = true
+      if (chosenReduction.monthsToExpiry !== null) {
+        if (Number(startMonth) > 0 && Number(startMonth) < 13) {
+          var startDate = new Date(Number(document.getElementById("start-year").value),
+                                   Number(document.getElementById("start-month").value),
+                                   Number(document.getElementById("start-day").value))
+          var endDate = new Date(startDate.setMonth(startDate.getMonth() + chosenReduction.monthsToExpiry))
+          readonly = true
+        }
       }
-    }
 
-    if (readonly){
-      document.getElementById('end-day').setAttribute('readonly', readonly)
-      document.getElementById('end-month').setAttribute('readonly', readonly)
-      document.getElementById('end-year').setAttribute('readonly', readonly)
-    } else {
-      document.getElementById('end-day').removeAttribute('readonly')
-      document.getElementById('end-month').removeAttribute('readonly')
-      document.getElementById('end-year').removeAttribute('readonly')
-    }
-    if(endDate !== null) {
-      document.getElementById('end-day').value = endDate.getDay()
-      document.getElementById('end-month').value = endDate.getMonth()
-      document.getElementById('end-year').value = endDate.getFullYear()
-    } else {
-      document.getElementById('end-day').value = ''
-      document.getElementById('end-month').value = ''
-      document.getElementById('end-year').value = ''
+      if (readonly){
+        document.getElementById('end-day').setAttribute('readonly', readonly)
+        document.getElementById('end-month').setAttribute('readonly', readonly)
+        document.getElementById('end-year').setAttribute('readonly', readonly)
+      } else {
+        document.getElementById('end-day').removeAttribute('readonly')
+        document.getElementById('end-month').removeAttribute('readonly')
+        document.getElementById('end-year').removeAttribute('readonly')
+      }
+      if(endDate !== null) {
+        document.getElementById('end-day').value = endDate.getDay()
+        document.getElementById('end-month').value = endDate.getMonth()
+        document.getElementById('end-year').value = endDate.getFullYear()
+      } else {
+        document.getElementById('end-day').value = ''
+        document.getElementById('end-month').value = ''
+        document.getElementById('end-year').value = ''
+      }
     }
+  }
+
+  window.onload = function () {
+    populateHours()
+    calculateExpiry()
```

Comment:
How come we are doing this on load? Surely the data we get back for the reduction should the source of truth?

Reply:
I was doing this more to set the hours and end date fields to read only,  should I change this to do only that?  Leave the actual values as they are in the returned reduction?